pennylane/compiler/python_compiler/dialects/quantum.py

```diff
@@ -792,7 +1365,37 @@ class SetStateOp(IRDLOperation):
 
 @irdl_op_definition
 class StateOp(IRDLOperation):
-    """Return the current statevector"""
+    """Return the current statevector.
+
+    The ``quantum.state`` operation represents the measurement process of returning the current
+    statevector in the computational basis. Typically reserved for simulator devices, although
+    in principle also achievable on hardware via tomography techniques.
+    While marginal states are supported, the operation is only well-defined if the provided
+    qubits are not entangled with the rest of the quantum state.
+    The result array contains one (complex) element for each possible bitstring, i.e. ``2^n`` where
+    ``n`` is the number of qubits.
+
+    Args:
+        obs (ObservableSSAValue):
```

Comment:
Let me know if this docstring seems reasonable:

```suggestion
        obs (ObservableSSAValue): An observable representing the qubits for which to
            extract the quantum state.
```